nengo/utils/numpy.py

```diff
@@ -108,6 +111,33 @@ def array_hash(a, n=100):
         return hash(v.data.tobytes())
 
 
+def array_hash(a, n=100):
+    """Simple fast array hash function.
+
+    For arrays with size larger than ``n``, pick ``n`` elements at random
+    to hash. This strategy should work well for dense matrices, but for
+    sparse ones it is more likely to give hash collisions.
```

Comment:
This comment needs updating. The thing about "sparse ones" was referring to numpy arrays with a lot of zeros. It doesn't apply to scipy sparse matrices.